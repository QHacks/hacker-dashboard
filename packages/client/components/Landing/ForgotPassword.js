import React, { Component } from "react";
import { Link } from "react-router-dom";
import * as constants from "../../assets/constants";
import ActionButton from "../ActionButton/ActionButton";
import Landing from "./Landing";

class ForgotPassword extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: ""
    };
  }
  render() {
    return (
      <Landing>
        <img
          src={"../../assets/img/qhacks-wordmark-colored.svg"}
          css={`
            max-height: 40px;
          `}
          alt="QHacks"
        />
        <h3
          css={`
            margin-top: 24px;
            color: ${constants.blue};
            font-weight: 700;
          `}
        >
          Forgot Password
        </h3>
        <p
          className="blurb"
          css={`
            line-height: 1.6;
            margin-top: 16px;
            color: #8a929f;
          `}
        >
          Please provide the email associated with your account to reset your
          password
        </p>
        <div
          css={`
            margin-top: 64px;
          `}
        >
          <input
            id="email"
            type="text"
            value={this.state.email}
            onChange={(e) => this.setState({ email: e.target.value })}
            placeholder="Enter your email address"
          />
        </div>
        <div
          css={`
            margin: 30px 0;
          `}
        >
          <Link className="landingLink" to="/login">
            Know you password? Login here!
          </Link>
        </div>
        <div>
          <ActionButton color="blue">Send reset link</ActionButton>
        </div>
      </Landing>
    );
  }
}

export default ForgotPassword;
