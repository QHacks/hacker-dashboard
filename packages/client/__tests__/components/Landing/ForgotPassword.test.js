import React from "react";
import ForgotPassword from "../../../components/Landing/ForgotPassword";
import { shallow } from "enzyme";

describe("ForgotPassword", () => {
  it("renders correctly", () => {
    expect(shallow(<ForgotPassword />)).toMatchSnapshot();
  });
});
