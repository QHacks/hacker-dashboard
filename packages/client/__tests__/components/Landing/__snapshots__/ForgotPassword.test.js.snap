// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPassword renders correctly 1`] = `
<Landing>
  <img
    alt="QHacks"
    className="css-zl7mak"
    src="../../assets/img/qhacks-wordmark-colored.svg"
  />
  <h3
    className="css-xqvf5y"
  >
    Forgot Password
  </h3>
  <p
    className="css-o4thmd blurb"
  >
    Please provide the email associated with your account to reset your password.
  </p>
  <div
    className="css-h8m1q6"
  >
    <input
      id="email"
      onChange={[Function]}
      placeholder="Enter your email address"
      type="text"
      value=""
    />
  </div>
  <div
    className="css-1tyn5zp"
  >
    <Link
      className="landingLink"
      replace={false}
      to="/login"
    >
      Know your password? Login here!
    </Link>
  </div>
  <div>
    <ActionButton
      color="blue"
      onClick={[Function]}
    >
      Send reset link
    </ActionButton>
  </div>
  <div
    className="css-17170mj"
  />
</Landing>
`;
